src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -394,6 +399,62 @@ yield return
                 }
             }
 
+            private StatementSyntax RemoveOutVariablesFromDeclarationExpressions(StatementSyntax statement,
+                IEnumerable<VariableInfo> variablesToRemove)
+            {
+                var replacements = new Dictionary<SyntaxNode, SyntaxNode>();
+
+                var declarations = statement.DescendantNodes()
+                    .Where(n => n.IsKind(SyntaxKind.DeclarationExpression, SyntaxKind.DeclarationPattern));
+
+                foreach (var node in declarations)
+                {
+                    switch (node.Kind())
+                    {
+                        case SyntaxKind.DeclarationExpression:
+                            var declaration = (DeclarationExpressionSyntax)node;
+                            if (declaration.VariableComponent.Kind() != SyntaxKind.TypedVariableComponent)
+                            {
+                                break;
+                            }
+
+                            var variableComponent = (TypedVariableComponentSyntax)declaration.VariableComponent;
+                            if (variableComponent.Designation.Kind() != SyntaxKind.SingleVariableDesignation)
+                            {
+                                break;
+                            }
+
+                            var designation = (SingleVariableDesignationSyntax)variableComponent.Designation;
+                            var name = designation.Identifier.ValueText;
+                            if (variablesToRemove.Select(v => v.Name).Contains(name))
```

Comment:
The list of variables that should be removed results from flow analysis on the block that will be extracted, so I don't think we'd have problems with duplicate variable names.
The rest of the refactoring code I saw only deals with name comparisons, not symbol comparisons. @CyrusNajmabadi What do you think?